File: RSA/python/RSA.py
from math import sqrt

def primeCheck(n):
    if n < 2:
        return False
    for i in range(2, int(sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def cal_gcd(a, b):
    if a == 0:
        return b
    else:
        return cal_gcd(b % a, a)

def cal_Euler_Totient_Function(p, q):
    return (p - 1) * (q - 1)

def key_generator():
    print("Enter 2 space seperated Prime Numbers")
    p, q = list(map(int, input().split()))
    if primeCheck(p) and primeCheck(q):
        #         print(p, q)
        n = p * q
        phi = cal_Euler_Totient_Function(p, q)

        e = 2
        while not cal_gcd(e, phi) == 1 and e < phi:
            e += 1
        print("e: ", e)

        d = 1
        while not (d * e) % phi == 1:
            d += 1

        print("d: ", d)

        return e, d, n
    else:
        print("Numbers entered are not prime. Exiting!!!")


def RSA():
    e, d, n = key_generator()
    plaintext = len(input("Enter the Text to be ecnrypted! \n"))
    ciphertext = (plaintext ** e) % n
    decryptedText = (ciphertext ** d) % n

    print("PlainText: {} \nCipherText: {} \nDecryptedText: {}".format(plaintext, ciphertext, decryptedText))

RSA()

File: RSA/C++/RSA.cpp
#include<iostream>
#include<math.h>
#include<vector>
using namespace std;

bool primeCheck(int n){
	if(n < 2){
		return false;
	}
	for(int i = 2; i < sqrt(n) + 1; i++){
		if(n % i == 0){
			return false;
		}
	}
	return true;
}

int gcd(int a, int b){
	if(a == 0){
		return b;
	}
	return gcd(b % a, a);
}

int cal_Euler_Totient_Function(int p, int q){
	return (p - 1) * (q - 1);
}

int KeyGenerator(){
	std::vector<int> vec;
	int p, q;

	cout<<"Enter two prime numbers!"<<endl;
	cin>>p>>q;

	if(!primeCheck(p) || !primeCheck(q)){
		exit(0);
	}

	int n = p * q;
	int phi = cal_Euler_Totient_Function(p, q);

	int k = 2, v = 1;

	while (!(gcd(k, phi) == 1)){

		k += 1;
	}

	cout<<"k: "<<k<<endl;

	while (!((k * v) % phi == 1)){
		v += 1;
	}

	vec[0] = k;
	vec[1] = v;
	vec[2] = n;

	return v;

}

int main(){

	int plainText, cipherText, decryptedText;

	cout<<"Enter Plain Text to be encrypted!"<<endl;
	cin>> plainText; 
	return 0;
}

